docs/tutorials/upgrading/upgrading-0.2-0.3.rst

```diff
@@ -0,0 +1,131 @@
+===========================
+Upgrading from v0.2 to v0.3
+===========================
+
+During the v0.3 release, a number of breaking changes were made that require you
+to update either deployment configuration and runtime configuration (e.g. Certificate,
+Issuer and ClusterIssuer resources).
+
+After reading these instructions, you should then proceed to upgrade cert-manager
+according to your deployment configuration (e.g. using ``helm upgrade`` if installing
+via Helm chart, or ``kubectl apply`` if installing with raw manifests).
+
+A brief summary:
+
+* Supporting resources for ClusterIssuer's (e.g. signing CA certificates, or
+  ACME account private keys) will now be stored in the same namespace as
+  cert-manager, instead of kube-system in previous versions (#329, @munnerz)
+
+* Switch to ConfigMaps instead of Endpoints for leader election (#327, @mikebryant)
+
+* Removing support for ACMEv1 in favour of ACMEv2 (#309, @munnerz)
+
+* Removing ingress-shim and compiling it into cert-manager itself (#502, @munnerz)
+
+* Change to the default behaviour of ingress-shim. It now generates Certificates that
+  with the ``ingressClass`` field set instead of the ``ingress`` field. This will
+  mean users of ingress controllers that assign a single IP to a single Ingress (e.g.
+  the GCE ingress controller) will no longer work without adding a new annotation
+  to your ingress resource.
+
+Supporting resources for ClusterIssuers moving into the cert-manager namespace
+==============================================================================
+
+Because ClusterIssuer resources do not reside in a namespace, when you depend upon
+supplemental resources (e.g. Secrets containing DNS provider credentials), we rely
+upon a 'service namespace' to store these resources.
+
+In the past, that namespace was hardcoded to ``kube-system``.
+With v0.3, we now store these resources in the same namespace as the cert-manager
+pod itself runs within.
+
+When upgrading, you should make sure to move any of these supplemental resources into
+the cert-manager deployment namespace, or otherwise deploy cert-manager into kube-system
+itself.
+
+You can also change the 'cluster-resource-namespace' with the ``--cluster-resource-namespace``
+flag, for example ``--cluster-resource-namespace=kube-system``.
+
+Switch to ConfigMaps instead of Endpoints for leader election
+=============================================================
+
+cert-manager-controller performs leader election to allow you to run 'hot standby'
+replicas of cert-manager.
+
+In the past, we used Endpoint resources to perform this election.
+The new best practice is to use ConfigMap resources in order to reduce API overhead
+in large clusters.
+
+As such, v0.3 switches us to use ConfigMap resources for leader election.
+
+During the upgrade, you should first scale your cert-manager-controller deployment
+to 0 to ensure no other replicas of cert-manager are running when the new v0.3
+deployment starts.
```

Comment:
Does the helm chart take care of this?
And if not using helm, is the user expected to modify the cert-manager image in the existing deployment pod template or should they create a new deployment after first scaling down and removing the previous deployment?

Reply:
Does the user need to clean up old supplemental resources? Endpoints etc?

Reply:
See #547, and the changes I've made to this area